Lazygit crashes on startup with panic index out of range

**Describe the bug**
The following is the result of attempting to launch lazygit inside a git repo

❯ lazygit
panic: runtime error: index out of range [1] with length 1

goroutine 21 [running]:
github.com/jesseduffield/lazygit/pkg/commands.(*BranchListBuilder).obtainBranches(0xc00013f6f0, 0x0, 0x0, 0x0)
        /private/tmp/lazygit-20210605-59301-6hnv7l/lazygit-0.28.2/pkg/commands/loading_branches.go:56 +0x6c6
github.com/jesseduffield/lazygit/pkg/commands.(*BranchListBuilder).Build(0xc0000beef0, 0x0, 0x9c11778, 0x10)
        /private/tmp/lazygit-20210605-59301-6hnv7l/lazygit-0.28.2/pkg/commands/loading_branches.go:97 +0x45
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshBranches(0xc0000ec000)
        /private/tmp/lazygit-20210605-59301-6hnv7l/lazygit-0.28.2/pkg/gui/branches_panel.go:71 +0xbe
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshCommits.func1()
        /private/tmp/lazygit-20210605-59301-6hnv7l/lazygit-0.28.2/pkg/gui/commits_panel.go:85 +0x46
github.com/jesseduffield/lazygit/pkg/utils.Safe.func1(0x1006e75, 0xc0000704e0)
        /private/tmp/lazygit-20210605-59301-6hnv7l/lazygit-0.28.2/pkg/utils/utils.go:107 +0x24
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0xc00013f7b8, 0x0, 0x0)
        /private/tmp/lazygit-20210605-59301-6hnv7l/lazygit-0.28.2/pkg/utils/utils.go:118 +0x62
github.com/jesseduffield/lazygit/pkg/utils.Safe(0xc0000a62d0)
        /private/tmp/lazygit-20210605-59301-6hnv7l/lazygit-0.28.2/pkg/utils/utils.go:107 +0x49
created by github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshCommits
        /private/tmp/lazygit-20210605-59301-6hnv7l/lazygit-0.28.2/pkg/gui/commits_panel.go:82 +0xa8

System details:
OS : macOS Big Sur 11.4
Terminal : iterm2 Build 3.4.8
Git : version 2.32.0
Homebrew : 3.2.2

Updated lazygit using $homebrew upgrade lazygit

lazygit: stable 0.28.2 (bottled)
Simple terminal UI for git commands
https://github.com/jesseduffield/lazygit/
/usr/local/Cellar/lazygit/0.28.2 (5 files, 16.3MB) *
  Poured from bottle on 2021-06-05 at 17:41:04
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/lazygit.rb
License: MIT
==> Dependencies
Build: go ✔
==> Analytics
install: 1,452 (30 days), 6,520 (90 days), 28,704 (365 days)
install-on-request: 1,452 (30 days), 6,519 (90 days), 28,671 (365 days)
build-error: 0 (30 days)

Similar problems on Fedora 34, both latest packaged version and source-built all the way back to lazygit 0.26. This is with a particular repository - several others seem to work fine, and this one worked fine last time I checked a couple of weeks ago. 

I'll be happy to provide more details if someone gives a hint as to what I should be looking for.

```
$ go version
go version go1.16.5 linux/amd64
$ lazygit
panic: runtime error: slice bounds out of range [:2] with length 0

goroutine 32 [running]:
github.com/jesseduffield/lazygit/pkg/commands.(*GitCommand).GetStatusFiles(0xc000097b00, 0x0, 0x0, 0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/commands/loading_files.go:39 +0xa8b
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshStateFiles(0xc0000f01c0, 0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/gui/files_panel.go:525 +0xb5
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshFilesAndSubmodules(0xc0000f01c0, 0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/gui/files_panel.go:112 +0xd4
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshSidePanels.func1.2.1()
	/builddir/build/BUILD/lazygit-0.28.2/pkg/gui/view_helpers.go:129 +0x2a
github.com/jesseduffield/lazygit/pkg/utils.Safe.func1(0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/utils/utils.go:107 +0x24
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0xc00044a7b8, 0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/utils/utils.go:118 +0x62
github.com/jesseduffield/lazygit/pkg/utils.Safe(0xc0004846a0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/utils/utils.go:107 +0x49
created by github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshSidePanels.func1.2
	/builddir/build/BUILD/lazygit-0.28.2/pkg/gui/view_helpers.go:129 +0x77
```

BTW running lazygit in a non-repository dir and answering N results in the same:

```
$ lazygit
Not in a git repository. Create a new git repository? (y/n): n
panic: runtime error: slice bounds out of range [:2] with length 0

goroutine 42 [running]:
github.com/jesseduffield/lazygit/pkg/commands.(*GitCommand).GetStatusFiles(0xc000181ce0, 0x0, 0x0, 0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/commands/loading_files.go:39 +0xa8b
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshStateFiles(0xc0001c21c0, 0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/gui/files_panel.go:525 +0xb5
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshFilesAndSubmodules(0xc0001c21c0, 0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/gui/files_panel.go:112 +0xd4
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshSidePanels.func1.2.1()
	/builddir/build/BUILD/lazygit-0.28.2/pkg/gui/view_helpers.go:129 +0x2a
github.com/jesseduffield/lazygit/pkg/utils.Safe.func1(0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/utils/utils.go:107 +0x24
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0xc0000e2fb8, 0x0, 0x0)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/utils/utils.go:118 +0x62
github.com/jesseduffield/lazygit/pkg/utils.Safe(0xc00044bf30)
	/builddir/build/BUILD/lazygit-0.28.2/pkg/utils/utils.go:107 +0x49
created by github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshSidePanels.func1.2
	/builddir/build/BUILD/lazygit-0.28.2/pkg/gui/view_helpers.go:129 +0x77
```

--- a/pkg/commands/loading_branches.go
+++ b/pkg/commands/loading_branches.go
@@ -52,6 +52,12 @@ func (b *BranchListBuilder) obtainBranches() []*models.Branch {
 		}
 
 		split := strings.Split(line, SEPARATION_CHAR)
+		if len(split) != 4 {
+			// Ignore line if it isn't separated into 4 parts
+			// This is probably a warning message, for more info see:
+			// https://github.com/jesseduffield/lazygit/issues/1385#issuecomment-885580439
+			continue
+		}
 
 		name := strings.TrimPrefix(split[1], "heads/")
 		branch := &models.Branch{

